src/Lib/Managers/SelectedBoundManager.vala

```diff
@@ -310,4 +318,48 @@ public class Akira.Lib.Managers.SelectedBoundManager : Object {
 
         canvas.window.event_bus.set_focus_on_canvas ();
     }
+
+    /**
+     * Move the item based on the mouse click and drag event.
+     */
+    private void move_from_event (
+        Lib.Items.CanvasItem item, double event_x, double event_y) {
+        if (!initial_drag_item_registered) {
+            initial_drag_item_registered = true;
+            initial_drag_item_x = item.bounds.x1;
+            initial_drag_item_y = item.bounds.y1;
+        }
+
+        // Keep reset and delta values for future adjustments.
+
+        // Calculate values needed to reset to the original position
+        var reset_x = item.bounds.x1 - initial_drag_item_x;
+        var reset_y = item.bounds.y1 - initial_drag_item_y;
```

Comment:
Better avoid using the `item.bounds` since those don't account for grouped items like the Artboard, which the bounds are not representative of the actual size of the item (eg. if an Artboard child is outside the artboard boundaries it expands the Artboard bounds).
Use the `item.transform.x` and `item.transform.y` which are the coordinates coming from the `Transform` component, which takes care of properly returning the correct values.